### ISSUE TITLE
map.wcs_utils should support hgln_obs and hglt_obs WCS keywords

### ISSUE THREAD

**Issue by dstansby** [sunpy/sunpy#3700]
### Description
Currently the `solar_wcs_frame_mapping` in `sunpy.map.wcs_utils` only support custom observer and rsun attributes that are added by SunPy. It should be possible to support the `'hgln_obs'` and `'hglt_obs'` keywords directly, so the example below works fine.

### Expected behavior
See code below; I expected the observer coordinate to be set in the resulting coordinate frame.

### Actual behavior
The observer coordinate is not set.

### Steps to Reproduce
```python
from astropy.wcs.utils import wcs_to_celestial_frame
from astropy.wcs import WCS
from sunpy.util import MetaDict
import sunpy.coordinates.wcs_utils

meta = MetaDict([('wcsaxes', 2), ('crpix1', 512.5), ('crpix2', 512.5),
                 ('cdelt1', 1.953125), ('cdelt2', 1.953125),
                 ('cunit1', 'arcsec'), ('cunit2', 'arcsec'),
                 ('ctype1', 'HPLN-TAN'), ('ctype2', 'HPLT-TAN'),
                 ('crval1', 0.0), ('crval2', 0.0),
                 ('lonpole', 180.0), ('latpole', 0.0),
                 ('date-obs', '2020-01-01T00:00:00.000'),
                 ('hgln_obs', 0.0), ('hglt_obs', -2.942433009185766),
                 ('dsun_obs', 147098545612.676),
                 ('rsun_ref', 695700000.0), ('rsun_obs', 975.5185219424909)])
wcs = WCS(meta)
print(wcs.heliographic_observer)

frame = wcs_to_celestial_frame(wcs)
print(frame)
```
prints
```
WCS Keywords

Number of WCS axes: 2
CTYPE : 'HPLN-TAN'  'HPLT-TAN'  
CRVAL : 0.0  0.0  
CRPIX : 512.5  512.5  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : 0.0005425347222222222  0.0005425347222222222  
NAXIS : 0  0
<Helioprojective Frame (obstime=2020-01-01T00:00:00.000, rsun=695700.0 km, observer=None)>
```

### System Details

  - SunPy Version: 1.1.0
  - Astropy Version: 4.0
  - Python Version: 3.7
  - OS information: macOS 10.15.1

**Comment by dstansby** [comment]
Woops, I think this might be a dupe of #2910 - might be worth appending my example above to that issue though?

**Comment by Cadair** [comment]
This would also be fixed properly by #3643 and is indeed a dupe of #2910. Please x-post your example :smile:
